### PR TITLE
[0.4.1] Chore: reraise as click exception

### PR DESCRIPTION
**Pull request by Karrenbelt**
## Proposed changes

use `reraise_as_click_exception` to fill some pytest coverage gaps.

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: autonomy/cli/build_images.py [hunk 1]
```diff
@@ -54,23 +54,17 @@ def build_image(
...
-    try:
+    with reraise_as_click_exception(FileNotFoundError):
         if agent is None:
             service_dir = Path(service_dir or Path.cwd()).absolute()
             service = load_service_config(service_dir)
             agent = service.agent
-    except FileNotFoundError as e:
-        raise click.ClickException(
-            "Service configuration not found the current directory"
-        ) from e
```
Comment (Karrenbelt) [inline autonomy/cli/build_images.py:65]: this changes the error message here, had to adjust the check in the test. I don't think there is any loss of meaningful information whatsoever though.

File: tests/test_autonomy/test_cli/test_build_image.py [hunk 2]
```diff
@@ -113,4 +113,5 @@ def test_service_file_missing(
         result = self.run_cli()
 
         assert result.exit_code == 1, result.output
-        assert "Service configuration not found the current directory" in result.output
+        assert "No such file or directory: " in result.output
+        assert "service.yaml" in result.output
```
Comment (Karrenbelt) [inline tests/test_autonomy/test_cli/test_build_image.py:117]: minor adjustment in test required to accomodate

**Review by 0xArdi — APPROVED** [review]

**Comment by Karrenbelt** [comment]
Base: 94.31% // Head: 94.77% // Increases project coverage by +0.45% :tada:

**Comment by Karrenbelt** [comment]
- failures e2e - #1430 
    ```
    2022-10-04T11:37:36.3517760Z [53111]stdout>    raise ABCIAppInternalError("last timestamp is None")
    2022-10-04T11:37:36.3563864Z [53111]stdout>packages.valory.skills.abstract_round_abci.base.ABCIAppInternalError: internal error: last timestamp is None
    ```

File: autonomy/cli/publish.py [hunk 3]
```diff
@@ -51,5 +52,3 @@ def publish(
             publish_agent_package(click_context, registry, push_missing)
         else:
             raise FileNotFoundError("No package config found in this directory.")
-    except Exception as e:  # pylint: disable=broad-except  # pragma: nocover
-        raise click.ClickException(str(e)) from e
```
Comment (Karrenbelt) [inline autonomy/cli/publish.py:55]: Here `FileNotFoundError` will also be reraised as a `ClickException`

File: autonomy/cli/build_images.py (hunk 1 above)
Comment (DavidMinarsch) [reply autonomy/cli/build_images.py:65]: Suggests that method is lacking an optional argument for a custom error message

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**